src/lib.rs

```diff
@@ -279,6 +282,12 @@ impl UnixStream {
         self.io.get_ref().peer_addr()
     }
 
+    /// Returns effective credentials of the process which called `connect` or `socketpair`.
+    #[cfg(any(target_os = "linux", target_os = "macos"))]
+    pub fn peer_cred(&self) -> io::Result<UCred> {
```

Comment:
OK, I'll go with ungated. Gating pushes the compilation error to consumer, so there'd still be the error. If you have some explanation about why compile error in this crate is better than at consumer, I'm interested in learning it. (I personally don't mind using the one you choose.)